### PR TITLE
Search results page

### PR DESCRIPTION
**Pull request by dajinchu**
added react router. home and results page separate. searchbar component abstracts controlling the input and hitting enter/the button to submit

TODO:
- [x] Delete old home.js
- [x] mobile styles

### CODE REVIEW

File: frontend/components/ResultsPage/ResultsFooter.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import FeedbackModal from '../FeedbackModal';
+
+export default function ResultsFooter() {
+  const [modalOpen, setModalOpen] = useState(false);
+
+  const toggleModal = () => {
+    setModalOpen(!modalOpen);
+  };
+  return (
+    <>
```
Comment (NEUDitao) [inline frontend/components/ResultsPage/ResultsFooter.js]: Why is there an empty boy here?

File: frontend/components/ResultsPage/ResultsFooter.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      <div className='footer'>
+
+        <div className='footer ui basic center aligned segment'>
+          See an issue or want to add to this website? Fork it or create an issue on
```
Comment (NEUDitao) [inline frontend/components/ResultsPage/ResultsFooter.js]: After talking last night -- should we add some sort of google form/something along those lines for people to sign up to help if they want to?

File: frontend/components/ResultsPage/ResultsFooter.js (hunk 1 above)
Comment (amielmon) [reply frontend/components/ResultsPage/ResultsFooter.js]: Fragment to group all children under single root node

File: frontend/components/pages/Home.js [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+  },
+  {
+    text: 'Spring 2019',
+    value: '201930',
+  },
+];
+
+const AVAILABLE_TERMS = termDropDownOptions.map((t) => { return t.value; });
+
+export default function Home() {
```
Comment (NEUDitao) [inline frontend/components/pages/Home.js:29]: Should we have the "SearchNEU is a sandbox project w/ awesome contributers (maybe just list like, Eddie, Jennings, Sean, etc), w/ all of the Feedabck /Report a bug/contact shtuffs?

Comment (dajinchu) [reply frontend/components/pages/Home.js:29]: yeah it is there

Comment (dajinchu) [reply frontend/components/pages/Home.js:29]: part of Footer

Comment (ryanhugh) [reply frontend/components/pages/Home.js:29]: Don't think you need to put my brother down anymore :)

File: frontend/components/App.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        </Route>
+        <Route path='/:termId?'>
+          <Home />
+        </Route>
+      </Switch>
+      <Route
+        path='/'
+        render={ ({ location }) => {
+          if (typeof window.ga === 'function') {
+            // Not sure if this works 100% right
```
Comment (ryanhugh) [inline frontend/components/App.js]: Basically, what do you what to count as a pageview in Google Analytics? Only when the user first goes to the site and the scripts in the <head> load? Or every time the user searches for something? Something else?

GA was designed back in the day when no websites were single page apps - it by default measures a pageview when its script first run on the page. 

With this site, code in the header will only run once the entire time the user is on the site, so if you count a pageview as when a user sees a different set of search results, you need to fire additional pings manually.

File: frontend/components/App.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+            window.ga('set', 'page', location.pathname + location.search);
```
Comment (ryanhugh) [inline frontend/components/App.js:28]: In the GA console, you can see what "pages" users are visiting. The "page" they are visiting is whatever data you send in the third argument here.

File: frontend/components/Home.js [hunk 6]
```diff
@@ -1,728 +0,0 @@
...
-// It calls into a bunch of different files
-// It calls into search.js and forward the results to resultsLoader
-// It also calls into FeedbackModal, if needed
-// And displays the SplashPage
-
-// Terms that used to be scraped at one point, but are no longer scraped.
-// These may be stored in user's localStorage from the last time they selected a term.
-// If they are, unselected them and select the latest term.
-const OLD_TERMS = [
-  '201858',
```
Comment (ryanhugh) [inline frontend/components/Home.js:33]: maybe for a future PR we can have these not hardcoded :)

File: frontend/components/images/logo.svg [hunk 7]
```diff
@@ -1,29 +1,12 @@
...
-    <path class="cls-2" d="M121.26,42.57h-1.4V19.26h2l7,19.69V19.26h1.4V42.57h-1.7l-7.29-20.79Z" transform="translate(-25.68 -18.89)" />
-    <path class="cls-2" d="M140.78,30v1.4h-6.36v9.73h7.66v1.4h-9.19V19.26h9.19v1.4h-7.66V30Z" transform="translate(-25.68 -18.89)" />
-    <path class="cls-2" d="M146,36.94c0,2.5.93,4.43,3.5,4.43s3.5-1.93,3.5-4.43V19.26h1.43V36.88c0,3.33-1.4,5.9-5,5.9s-5-2.57-5-5.9V19.26H146Z" transform="translate(-25.68 -18.89)" />
-</svg>
\ No newline at end of file
+<svg width="194" height="36" viewBox="0 0 194 36" fill="none" xmlns="http://www.w3.org/2000/svg">
+<path d="M8.1042 0C13.3498 0 16.0611 3.13575 16.0611 8.62698V9.70167H10.948V8.22949C10.948 5.77095 9.9608 4.84348 8.25155 4.84348C6.5423 4.84348 5.55506 5.77095 5.55506 8.22949C5.55506 15.296 16.1053 16.6209 16.1053 26.4256C16.1053 31.9169 13.3498 35.0526 8.06 35.0526C2.77016 35.0526 0 31.9611 0 26.4993V24.3646H5.09828V26.8231C5.09828 29.2817 6.17393 30.1503 7.89791 30.1503C9.6219 30.1503 10.6975 29.267 10.6975 26.8231C10.6975 19.7567 0.147349 18.4317 0.147349 8.62698C0.147349 3.12102 2.84384 0 8.1042 0Z" fill="#D41B2C"/>
```
Comment (ryanhugh) [inline frontend/components/images/logo.svg:2]: Can we keep the AGPL3 licence at the top of every file? You probably already know why we have that licence, but PM me if you wanna hear the reasons

File: frontend/components/TopSearchBar.js [hunk 8]
```diff
@@ -0,0 +1,56 @@
+import React, { useState } from 'react';
```
Comment (ryanhugh) [inline frontend/components/TopSearchBar.js]: Can we keep the AGPL3 licence at the top of every file? You probably already know why we have that licence, but PM me if you wanna hear the reasons

File: frontend/components/pages/Home.js [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+        <p>
+          <a href='https://forms.gle/HNJ1AWTCXnu3XovKA' target='_blank' rel='noopener noreferrer' className='getInvolvedText'>
+            Take our survey
+          </a>
+        </p>
+      </div>
+    );
+  }
+
+  // Not totally sure why, but this height: 100% removes the extra whitespace at the bottom of the page caused by the upward translate animation.
```
Comment (ryanhugh) [inline frontend/components/pages/Home.js:62]: Are you guys keeping the transition animation? 

I can explain this comment some more if ya wnna hear

Comment (NEUDitao) [reply frontend/components/pages/Home.js:62]: I raised this in slack earlier today, they'll get it!

File: frontend/components/pages/Home.js (hunk 3 above)
Comment (NEUDitao) [reply frontend/components/pages/Home.js:29]: I don't see it on my machine on the home page..

File: frontend/components/pages/Home.js (hunk 9 above)
Comment (dajinchu) [reply frontend/components/pages/Home.js:62]: yeah the results will go blank during the request whenever a new search query is entered to make it clear that its a new result set

File: frontend/components/images/logo.svg (hunk 7 above)
Comment (dajinchu) [reply frontend/components/images/logo.svg:2]: yep

File: frontend/components/App.js (hunk 4 above)
Comment (dajinchu) [reply frontend/components/App.js]: yeah it works. App.js rerenders whenever we go to a new url, so the ga pageview will get sent

Comment (dajinchu) [reply frontend/components/App.js]: was afraid it was double logging, but put in a console.log to check and it seems good

File: frontend/components/pages/Results.js [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+import { Dropdown } from 'semantic-ui-react';
+import logo from '../images/logo.svg';
+import search from '../search';
+import macros from '../macros';
+import ResultsLoader from '../ResultsLoader';
+import SearchBar from '../ResultsPage/SearchBar';
+import Footer from '../Footer';
+
+
+const termDropDownOptions = [
```
Comment (NEUDitao) [inline frontend/components/pages/Results.js]: This appears twice -- is there some way we can consolidate it somewhere?

File: frontend/components/pages/Home.js [hunk 11]
```diff
@@ -0,0 +1,158 @@
...
+
+  const [searchFocused, setSearchFocused] = useState(false);
+
+  // On mobile only show the logo and the github corner if there are no results and the search box is not focused (the virtual keyboard is not on the screen).
+  let containerClassnames = 'home-container';
+  if (macros.isMobile && searchFocused) {
+    containerClassnames += ' mobileCompact';
+  }
+
+  const wantToHelpOpacity = 1;
```
Comment (NEUDitao) [inline frontend/components/pages/Home.js]: why is this a number? -- it seems to be just doing what a boolean flag would be

File: frontend/components/pages/Results.js (hunk 10 above)
Comment (dajinchu) [reply frontend/components/pages/Results.js]: fixed

File: frontend/components/pages/Home.js (hunk 11 above)
Comment (dajinchu) [reply frontend/components/pages/Home.js]: it was in the old code lol. fixed

**Pull request merged**